validator/client/beacon-api/duties.go

```diff
@@ -45,3 +252,45 @@ func (c beaconApiDutiesProvider) GetAttesterDuties(ctx context.Context, epoch ty
 
 	return attesterDuties.Data, nil
 }
+
+// GetProposerDuties retrieves the proposer duties for the given epoch
+func (c beaconApiDutiesProvider) GetProposerDuties(ctx context.Context, epoch types.Epoch) ([]*apimiddleware.ProposerDutyJson, error) {
+	proposerDuties := apimiddleware.ProposerDutiesResponseJson{}
+	if _, err := c.jsonRestHandler.GetRestJsonResponse(ctx, fmt.Sprintf("/eth/v1/validator/duties/proposer/%d", epoch), &proposerDuties); err != nil {
+		return nil, errors.Wrapf(err, "failed to query proposer duties for epoch `%d`", epoch)
+	}
+
+	for index, proposerDuty := range proposerDuties.Data {
```

Comment:
You may want to check if `proposerDuties.Data == nil` first, like you do in `GetCommittees`.